org.eclipse.m2e.profiles.core/src/org/eclipse/m2e/profiles/core/internal/management/ProfileManager.java

```diff
@@ -248,9 +248,12 @@ private String findSource(Profile profile, List<Model> modelHierarchy) {
   private RepositorySystem getRepositorySystem() {
     try {
       //TODO find an alternative way to get the Maven RepositorySystem, or use Aether directly to resolve models??
-      return MavenPluginActivator.getDefault().getPlexusContainer().lookup(RepositorySystem.class);
-    } catch(ComponentLookupException e) {
-      throw new NoSuchComponentException(e);
+      return MavenPluginActivator.getDefault().getMaven().lookup(RepositorySystem.class);
+    } catch(CoreException e) {
+      if(e.getStatus().getException() instanceof ComponentLookupException) {
+        throw new NoSuchComponentException((ComponentLookupException) e.getStatus().getException());
+      }
+      throw new NoSuchComponentException(null);
```

Comment:
we completely lose the exception information/context that way. At least log the cause of the failure